### PR TITLE
Add a one time retry of license failure

### PR DESCRIPTION
**Pull request by eivindjahren**
To combat unreliable license server issues. Add a simple retry to ecl_run.

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by berland — APPROVED** [review]

File: tests/unit_tests/shared/share/test_ecl_run.py [hunk 1]
```diff
@@ -564,3 +565,43 @@ def test_slurm_env_parsing():
         "ws8",
         "ws8",
     ]
+
+
+def test_ecl100_retries_on_license_failure(tmp_path, monkeypatch):
```
Comment (berland) [inline tests/unit_tests/shared/share/test_ecl_run.py:570]: `retries_once_`

File: tests/unit_tests/shared/share/test_ecl_run.py [hunk 2]
```diff
@@ -564,3 +565,43 @@ def test_slurm_env_parsing():
...
+        stat.S_IEXEC | stat.S_IWUSR | mock_eclipse_path.stat().st_mode
+    )
+    monkeypatch.setenv("ECL100_SITE_CONFIG", str(tmp_path / "mock_config.yaml"))
+    econfig = ecl_config.Ecl100Config()
+    sim = econfig.sim("2015.2")
+    erun = ecl_run.EclRun(str(case_path), sim)
+
+    with pytest.raises(RuntimeError, match="LICENSE FAILURE"):
+        erun.runEclipse()
+    assert (tmp_path / "mock_log").read_text() == "Called mock\nCalled mock\n"
```
Comment (berland) [inline tests/unit_tests/shared/share/test_ecl_run.py:607]: Assert `== "Called mock\n" * max_attempts` and set` max_attempts = 2` above (too make it more obvious?)

File: tests/unit_tests/shared/share/test_ecl_run.py (hunk 1 above)
Comment (eivindjahren) [reply tests/unit_tests/shared/share/test_ecl_run.py]: Fixed

File: tests/unit_tests/shared/share/test_ecl_run.py (hunk 2 above)
Comment (eivindjahren) [reply tests/unit_tests/shared/share/test_ecl_run.py]: Fixed

File: tests/unit_tests/shared/share/test_ecl_run.py [hunk 3]
```diff
@@ -564,3 +566,47 @@ def test_slurm_env_parsing():
         "ws8",
         "ws8",
     ]
+
+
+@pytest.mark.skipif(
+    not sys.platform.startswith("darwin"), reason="Flaky bash mock on mac"
```
Comment (berland) [inline tests/unit_tests/shared/share/test_ecl_run.py:572]: `not`?

**Review by berland — APPROVED** [review]
:+1:

**Pull request merged**